pkg/sink/sink.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sink
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"path"
+	"strings"
+
+	triggersclientset "github.com/tektoncd/triggers/pkg/client/clientset/versioned"
+	"github.com/tektoncd/triggers/pkg/template"
+	"github.com/tidwall/gjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientset "k8s.io/client-go/kubernetes"
+)
+
+type Resource struct {
+	TriggersClient         *triggersclientset.Clientset
+	KubeClient             *kubeclientset.Clientset
+	EventListenerName      string
+	EventListenerNamespace string
+}
+
+func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Request) {
+	el, err := r.TriggersClient.TektonV1alpha1().EventListeners(r.EventListenerNamespace).Get(r.EventListenerName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting EventListener %s in Namespace %s: %s", r.EventListenerName, r.EventListenerNamespace, err)
+		return
+	}
+
+	event, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		log.Printf("Error reading event body: %s", err)
+		return
+	}
+
+	for _, tbRef := range el.Spec.TriggerBindingRefs {
+		tbNamespace := tbRef.Namespace
+		if tbNamespace == "" {
+			tbNamespace = r.EventListenerNamespace
+		}
+		tb, err := r.TriggersClient.TektonV1alpha1().TriggerBindings(tbNamespace).Get(tbRef.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("Error getting TriggerBinding %s in Namespace %s: %s", tbRef.Name, tbNamespace, err)
+			continue
```

Comment:
Why are we just logging error and then continuing the iteration? Why not stop the iteration? Similar approach is being used below.

Reply:
I was doing this because the EventListener has a list of TriggerBindings which I thought should act independently of one another. Just because one of the TriggerBindings fails, it doesn't necessarily mean that the others will fail.

Reply:
Got it. Thanks for the clarification.